libV2/schemaUtils.js

```diff
@@ -0,0 +1,1620 @@
+      example = resolveExampleData(context, example.value);
+    }
+
+    return example;
+  },
+
+  /**
+   * Handle resoltion of allOf property of schema
+   *
+   * @param {Object} context - Global context object
+   * @param {Object} schema - Schema to be resolved
+   * @param {Number} [stack] - Current recursion depth
+   * @param {*} resolveFor - resolve refs for flow validation/conversion (value to be one of VALIDATION/CONVERSION)
+   * @param {Object} seenRef - Map of all the references that have been resolved
+   *
+   * @returns {Object} Resolved schema
+   */
+  resolveAllOfSchema = (context, schema, stack, resolveFor = CONVERSION, seenRef = {}) => {
+    try {
+      return mergeAllOf(_.assign(schema, {
+        allOf: _.map(schema.allOf, (schema) => {
+          // eslint-disable-next-line no-use-before-define
+          return resolveSchema(context, schema, stack, resolveFor, _.cloneDeep(seenRef));
+        })
+      }), {
+        resolvers: {
+          // for keywords in OpenAPI schema that are not standard defined JSON schema keywords, use default resolver
+          defaultResolver: (compacted) => { return compacted[0]; }
+        }
+      });
+    }
+    catch (e) {
+      console.warn('Error while resolving allOf schema: ', e);
+      return { value: '<Error: Could not resolve allOf schema' };
+    }
+  },
+
+  /**
+   * Resolve a given ref from the schema
+   *
+   * @param {Object} context - Global context
+   * @param {Object} schema - Schema that is to be resolved
+   * @param {Number} [stack] - Current recursion depth
+   * @param {String} resolveFor - For which action this resoltion is to be done
+   * @param {Object} seenRef - Map of all the references that have been resolved
+   *
+   * @returns {Object} Returns the object that staisfies the schema
+   */
+  resolveSchema = (context, schema, stack = 0, resolveFor = CONVERSION, seenRef = {}) => {
+    if (!schema) {
+      return new Error('Schema is empty');
+    }
+
+    if (stack >= REF_STACK_LIMIT) {
+      return { value: ERR_TOO_MANY_LEVELS };
+    }
+
+    stack++;
+
+    const compositeSchema = schema.anyOf || schema.oneOf,
+      compositeKeyword = schema.anyOf ? 'anyOf' : 'oneOf',
+      { concreteUtils } = context;
+
+    if (compositeSchema) {
+      if (resolveFor === CONVERSION) {
+        return resolveSchema(context, compositeSchema[0], stack, resolveFor, _.cloneDeep(seenRef));
+      }
+
+      return { [compositeKeyword]: _.map(compositeSchema, (schemaElement) => {
+        PROPERTIES_TO_ASSIGN_ON_CASCADE.forEach((prop) => {
+          if (_.isNil(schemaElement[prop]) && !_.isNil(schema[prop])) {
+            schemaElement[prop] = schema[prop];
+          }
+        });
+        return resolveSchema(context, schemaElement, stack, resolveFor, _.cloneDeep(seenRef));
+      }) };
+    }
+
+    if (schema.allOf) {
+      return resolveAllOfSchema(context, schema, stack, resolveFor, _.cloneDeep(seenRef));
+    }
+
+    if (schema.$ref) {
```

Comment:
@barshan23 why is this condition here? This should be handled in the orchestrator which is recursing this function right?